[refactor]: introduce full-featured code groups (#224)
hyperledger/iroha-2-docs: src/documenting/code-groups.md

# Code Groups

Iroha 2 Documentation includes a custom Markdown Extension in order to
support code groups. It is achieved with `:::code-group` container.

## Markdown Syntax

**Input**

````md
:::code-group

```ts
type A = [string, string]
```

### I have a title

```rust
assert_eq!(vec![], vec![]);
```

```
Fence without lang or title
```

### Snippet syntax is supported as well

<<<@/snippets/debug_java_Lorem.java

:::
````

**Output**

:::code-group

```ts
type A = [string, string]
```

### I have a title

```rust
assert_eq!(vec![], vec![]);
```

```
Fence without lang or title
```

### Snippet syntax is supported as well

<<<@/snippets/debug_java_Lorem.java

:::

> **Rules:**

Reviewer: I'd put this before input-output example

Author: Done

> - All fences (i.e. code blocks) within a `code-group` container are
>   considered as dedicated tabs;

Reviewer: ```suggestion
- Each fence (a code block) within a `code-group` container is
  rendered as a tab;
```

> - If fence has a heading before it, then the heading content is used as a

Reviewer 2: ```suggestion
- If the fence has a heading before it, then the heading content is used as a
```

> - If the fence has a heading before it, then the heading content is used as a

Reviewer: ```suggestion
- If a fence has a heading before it, then the heading content is used as a
```

>   title. Heading could have any level (i.e. `#`, `##` and so on).

Reviewer 2: ```suggestion
  title. The heading could have any level (i.e. `#`, `##`, and so on).
```

> - If fence doesn't have a title, then it's language is used as a title. If

Reviewer 2: ```suggestion
- If the fence doesn't have a title, then it's language is used as a title. If
```

> - If the fence doesn't have a title, then it's language is used as a title. If

Reviewer: ```suggestion
- If a fence doesn't have a title, then its language is used as a title. If
```

>   there is no language, then titles fallbacks to `<block-#i>` placeholder.

Reviewer 2: ```suggestion
  there is no language, then the title falls back to the `<block-#i>` placeholder.
```

## HTML-based syntax

> It is more low-level, but might be used for advanced cases.

Reviewer 2: ```suggestion
It is lower-level but might be used for advanced cases.
```
